src/erlcloud_inspector.erl

```diff
@@ -0,0 +1,1155 @@
+      }.
+
+-spec new(string(), string(), string()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey,
+       inspector_host=Host
+      }.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{
+       access_key_id=AccessKeyID,
+       secret_access_key=SecretAccessKey,
+       inspector_host=Host,
+       inspector_port=Port
+      }.
+
+-spec configure(string(), string()) -> ok.
+
+configure(AccessKeyID, SecretAccessKey) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey)),
+    ok.
+
+-spec configure(string(), string(), string()) -> ok.
+
+configure(AccessKeyID, SecretAccessKey, Host) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host)),
+    ok.
+
+-spec configure(string(), string(), string(), non_neg_integer()) -> ok.
+
+configure(AccessKeyID, SecretAccessKey, Host, Port) ->
+    put(aws_config, new(AccessKeyID, SecretAccessKey, Host, Port)),
+    ok.
+
+default_config() ->
+    erlcloud_aws:default_config().
+
+
+-spec inspector_result_fun/1 ::
+          (Request :: aws_request()) -> aws_request().
+inspector_result_fun(#aws_request{response_type = ok} = Request) ->
+    Request;
+inspector_result_fun(#aws_request{response_type = error,
+                                  error_type = aws,
+                                  response_status = Status} = Request) when Status >= 500 ->
+    Request#aws_request{should_retry = true};
+inspector_result_fun(#aws_request{response_type = error, error_type = aws} = Request) ->
+    Request#aws_request{should_retry = false}.
+
+%%%------------------------------------------------------------------------------
+%%% Inspector API Functions
+%%%------------------------------------------------------------------------------
+
+
+%%%------------------------------------------------------------------------------
+%%% AddAttributesToFindings
+%%
+%% http://docs.aws.amazon.com/inspector/latest/APIReference/API_AddAttributesToFindings.html
+%%%------------------------------------------------------------------------------
+-type attribute() :: proplists:proplist().
+
+-spec add_attributes_to_findings/2 ::
+          (Attributes :: [attribute()],
+           FindingArns :: [string()]) ->
+          return_val().
+add_attributes_to_findings(Attributes, FindingArns) ->
+    add_attributes_to_findings(Attributes, FindingArns, default_config()).
+
+
+-spec add_attributes_to_findings/3 ::
+          (Attributes :: [attribute()],
+           FindingArns :: [string()],
+           Config :: aws_config()) ->
+          return_val().
+add_attributes_to_findings(Attributes, FindingArns, Config) ->
+    Json = [{<<"attributes">>, Attributes},
+            {<<"findingArns">>, [to_binary(A) || A <- FindingArns]}],
+    inspector_request(Config, "InspectorService.AddAttributesToFindings", Json).
```

Comment:
i noticed that this module has `"InspectorService.*"` prefix for all operations. Are SignV4 headers okay with this as rest of the module have just operation without prefix?